drivers/platform/stm32/stm32_dma.c

```diff
@@ -68,7 +69,11 @@ int stm32_dma_config_xfer(struct no_os_dma_ch *channel,
 
 	sdma_ch = channel->extra;
 
+#if defined (STM32L4) || defined (STM32H5)
```

Comment:
a comment here would be nice... why only these families? I see there are other families which have hdma->Instance available. I think the problem is that the Init.Channels is not available for most of them.
so maybe change this check to only use Init.Channel for stm32f2, f4 and f7: https://github.com/search?q=org%3ASTMicroelectronics+hdma-%3EInit.Channel&type=code
also, this change is not related to this commit and should be part of the first commit.